test/helpers/generateConfig.js

```diff
@@ -0,0 +1,40 @@
+import getModuleFromPath from '../../helpers/getModuleFromPath';
+import generateConfig from '../../helpers/generateConfig';
+import interopRequireDefault from '../../helpers/interopRequireDefault';
+
+jest.mock('../../helpers/getModuleFromPath');
+getModuleFromPath.mockImplementation((path, options) => () => ({ path, options }));
+
+describe('generateConfig', () => {
+  let genConfigPromise;
+  const config = {
+    sync: { hooks: ['module1', 'module2', 'module3'] },
+  };
+  const moduleConfig = {
+    projectRoot: process.cwd(),
+    requireInteropWrapper: interopRequireDefault,
+  };
+
+  beforeEach(() => {
+    genConfigPromise = generateConfig(config);
+  });
+
+  it('should trigger sync config module functions in order', () => {
+    const { hooks } = config.sync;
+
+    return genConfigPromise.then((result) => {
+      expect(Array.isArray(result)).toBe(true);
+
+      result.forEach(({ path }, index) => {
+        expect(path).toBe(hooks[index]);
+        expect(getModuleFromPath).toHaveBeenCalledWith(
+          hooks[index], moduleConfig,
+        );
+      });
+
+      expect(getModuleFromPath).toHaveBeenCalledTimes(3);
+    }).catch((err) => {
+      console.error(err);
```

Comment:
i don't think we want this catch; if it's omitted, a rejection will fail the test.

Reply:
oops yeah, this was here for my testing purposes, will remove.